src/internals/ir/callGraph.ts

```diff
@@ -52,64 +70,89 @@ class CGNode {
       this.logger.debug(`CGNode created without AST ID for function "${name}"`);
     }
   }
+
+  // Method to set a flag
+  public setFlag(flag: number) {
+    this.flags |= flag;
+  }
+
+  // Method to check if a flag is set
+  public hasFlag(flag: number): boolean {
+    return (this.flags & flag) !== 0;
+  }
 }
 
 /**
- * The `CallGraph` class represents a directed graph where nodes correspond to functions
- * or methods in a program, and edges indicate calls between them.
+ * Represents the call graph, a directed graph where nodes represent functions or methods,
+ * and edges indicate calls between them.
  *
- * Nodes and edges are uniquely identified using indices generated by `IdxGenerator`.
- * This class provides methods to construct the graph from AST data, retrieve nodes and edges,
- * and check connectivity between nodes.
+ * The `CallGraph` class provides methods to build the graph from a Tact AST, retrieve nodes/edges,
+ * analyze connectivity, and add function calls dynamically.
  */
 export class CallGraph {
   private nodeMap: Map<CGNodeId, CGNode> = new Map();
-  private edgesMap: Map<CGEdgeId, CGEdge> = new Map();
+  private astIdToNodeId: Map<number, CGNodeId> = new Map();
```

Comment:
```suggestion
  private astIdToNodeId: Map<AstNode["id"], CGNodeId> = new Map();
```

Correct?

Reply:
No, we cant remove edgesMap.  Part of logic.